src/Simulation/CsharpGeneration/SimulationCode.fs

```diff
@@ -93,9 +93,7 @@ module SimulationCode =
     let isGeneric context (n: QsQualifiedName) =
         if context.allCallables.ContainsKey n then
             let signature = context.allCallables.[n].Signature
-            let tIn = signature.ArgumentType
-            let tOut = signature.ReturnType
-            hasTypeParameters [tIn;tOut]
```

Comment:
do we still need the `hasTypeParameters` method?

Reply:
The hasTypeParameters method is still used in some other parts of the code in this file, which doesn't seem to be related to the bug.

Reply:
It can still be useful for determining if a given list of types contains references to type parameters.